tsl/test/expected/continuous_aggs_bgw.out

```diff
@@ -154,28 +150,160 @@ SELECT * FROM _timescaledb_config.bgw_job where id=:job_id;
---------+------------------------------+------------------------------+------------------+------------+-----------------+----------------+---------------
-   1000 | Sat Jan 01 04:00:00 2000 PST | Fri Dec 31 16:00:00 1999 PST | t                |          1 |               1 |              0 |             0
+ job_id |          next_start          |         last_finish          |  ?column?  | last_run_success | total_runs | total_successes | total_failures | total_crashes 
+--------+------------------------------+------------------------------+------------+------------------+------------+-----------------+----------------+---------------
+   1000 | Sat Jan 01 04:00:00 2000 PST | Fri Dec 31 16:00:00 1999 PST | @ 12 hours | t                |          1 |               1 |              0 |             0
+(1 row)
+
+--clear log for next run of scheduler.
+TRUNCATE public.bgw_log;
+CREATE FUNCTION wait_for_timer_to_run(started_at INTEGER, spins INTEGER=:TEST_SPINWAIT_ITERS) RETURNS BOOLEAN LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+	num_runs INTEGER;
+	message TEXT;
+BEGIN
+	select format('[TESTING] Wait until %%, started at %s', started_at) into message;
+	FOR i in 1..spins
+	LOOP
+	SELECT COUNT(*) from bgw_log where msg LIKE message INTO num_runs;
+	if (num_runs > 0) THEN
+		RETURN true;
+	ELSE
+        RAISE WARNING 'waiting';
+		PERFORM pg_sleep(0.1);
+	END IF;
+	END LOOP;
+	RETURN false;
+END
+$BODY$;
+CREATE FUNCTION wait_for_job_to_run(job_param_id INTEGER, expected_runs INTEGER, spins INTEGER=:TEST_SPINWAIT_ITERS) RETURNS BOOLEAN LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+	num_runs INTEGER;
+BEGIN
+	FOR i in 1..spins
+	LOOP
+	SELECT total_successes FROM _timescaledb_internal.bgw_job_stat WHERE job_id=job_param_id INTO num_runs;
+	if (num_runs = expected_runs) THEN
+		RETURN true;
+    ELSEIF (num_runs > expected_runs) THEN
+        RAISE 'num_runs > expected';
+	ELSE
+		PERFORM pg_sleep(0.1);
+	END IF;
+	END LOOP;
+	RETURN false;
+END
+$BODY$;
+--make sure there is 1 job to start with
+SELECT wait_for_job_to_run(:job_id, 1);
+ wait_for_job_to_run 
+---------------------
+ t
+(1 row)
+
+SELECT ts_bgw_params_mock_wait_returns_immediately(:WAIT_FOR_OTHER_TO_ADVANCE);
+ ts_bgw_params_mock_wait_returns_immediately 
+---------------------------------------------
+ 
+(1 row)
+
+--start the scheduler on 0 time
+SELECT ts_bgw_params_reset_time(0, true);
+ ts_bgw_params_reset_time 
+--------------------------
+ 
+(1 row)
+
+SELECT ts_bgw_db_scheduler_test_run(24*60*60*1000, 0);
+ ts_bgw_db_scheduler_test_run 
+------------------------------
+ 
+(1 row)
+
+SELECT wait_for_timer_to_run(0);
+WARNING:  waiting
+ wait_for_timer_to_run 
+-----------------------
+ t
+(1 row)
+
+--advance to 12:00 so that it runs one more time; now we know the
+--scheduler has loaded up the job with the old schedule_interval
+SELECT ts_bgw_params_reset_time(12::bigint*60*60*1000*1000, true);
```

Comment:
I kind of wish these numbers were a little less magic. I assume this is
`12 * :MINUTES_PER_HOUR * :SECONDS_PER_MINUTE * :MS_PER_S * :US_PER_MS`?

Reply:
`12 * :M_P_H * :S_P_M * :MS_P_S * :US_P_MS` 🤔

Reply:
extract(epoch from interval '12 hour') * 1000000